src/NotificationTarget.php

```diff
@@ -222,6 +222,19 @@ public function validateSendTo($event, array $infos, $notify_me = false, $emitte
             }
         }
 
+        //do not notify if user explicitly refused it
+        //Except for CommonITILOject (already manage by use_notification)
+        $user = new User();
+        if (
+            is_string($itemtype)
+            && (!is_a($itemtype, CommonITILObject::class, true) || !is_a($itemtype, NotificationTargetCommonITILObject::class, true))
+            && (!is_a($itemtype, NotificationTargetUser::class, true))
```

Comment:
This still requires hardcoded cases here, and it will be a problem for plugins.

Maybe the solution is to add a new method.

```suggestion
            !$this->canNotificationBeDisabled($event)
```

```php
// NotificationTarget::canNotificationBeDisabled()
protected function canNotificationBeDisabled(string $event): bool
{
    return true;
}

// NotificationTargetCommonITILObject::canNotificationBeDisabled()
protected function canNotificationBeDisabled(string $event): bool
{
    // Notifications on ITIL objects are relying on `use_notification` property of actors.
    return false;
}

// NotificationTargetUser::canNotificationBeDisabled()
protected function canNotificationBeDisabled(string $event): bool
{
    if (in_array($event, ['passwordinit', 'passwordforget', 'passwordexpires'])) {
        // Notifications related to passwords should never be disabled.
        return false;
    }

    return true;
}
```

Reply:
With this, changes made to `validateSendTo()` in other classes will not be required anymore.

Reply:
From ```NotificationTarget```, ``$event``` is not like ```passwordinit``` / ```new```, ```update``` etc ...

Ir refere to```NotificationEventMailing``` for example

Reply:
:+1:

move to dedicated method done!